managed/services/minio/service.go

```diff
@@ -111,7 +124,80 @@ func (s *Service) RemoveRecursive(ctx context.Context, endpoint, accessKey, secr
 	return nil
 }
 
-func newClient(endpoint, accessKey, secretKey string) (*minio.Client, error) {
+// List provides an abstraction over the minio API to list all objects in the bucket
+// It scans path with prefix and returns all files with given suffix.
+// Both prefix and suffix can be omitted.
+func (m *Client) List(ctx context.Context, prefix, suffix string) ([]FileInfo, error) {
+	var files []FileInfo
+	if prefix != "" && !strings.HasSuffix(prefix, "/") {
+		prefix += "/"
+	}
+
+	var g errgroup.Group
+
+	g.Go(func() error {
```

Comment:
Why do we need error group here? I mean we starting goroutine and immediately start waiting it.